packages/server/src/controllers/Controller.js

```diff
@@ -533,3 +475,54 @@ export class Controller {
 EventEmitter.mixin(Controller.prototype)
 
 const inheritanceMap = new WeakMap()
+
+const setupHandlerFromObject = ({
+  handler,
+  action,
+  authorize,
+  parameters,
+  returns,
+  scope,
+  transacted
+}) => {
+  handler.authorize = authorize
+  handler.transacted = transacted
+
+  if (action) {
+    const [verb, path] = asArray(action)
+    handler.verb = verb
+    handler.path = path
+  }
+
+  if (parameters) {
+    const [_parameters, options] = parameters
+    const hasOptions = isArray(_parameters)
+    handler.parameters = hasOptions ? _parameters : parameters
+
+    // If validation options are provided, expose them through
+    // `handler.options.parameters`, see ControllerAction
+    if (hasOptions) {
+      handler.options = handler.options || {}
+      handler.options.parameters = options
```

Comment:
How about creating a new object instead of modifying the one that's passed in:
```js
handler.options = {
   …handler.options,
   parameters: options
}
```